src/main/java/com/aws/greengrass/shadowmanager/sync/RequestBlockingQueue.java

```diff
@@ -473,6 +474,26 @@ public int size() {
         }
     }
 
+    /**
+     * Returns true if the total capacity of the queue could be updated. Returns false if the capacity is less than
+     * the current number of requests.
+     *
+     * @param newCapacity number of requests to queue.
+     * @return true if the queue capacity could be updated, false if the desired capacity is too small.
+     */
+    public boolean updateCapacity(int newCapacity) {
+        lock.lock();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Lock acquisition without timeout detected. Using locks without timeouts can lead to  indefinite blocking and service outages if the lock cannot be acquired. Use timeout-based methods like tryLock(timeout), tryAcquire(timeout), or await(timeout) instead. Learn more on : https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/locks/Lock.html#tryLock-long-java.util.concurrent.TimeUnit